python/src/deltachat/direct_imap.py

```diff
@@ -224,9 +224,7 @@ def idle_check(self, terminate=False):
         """ (blocking) wait for next idle message from server. """
         assert self._idling
         self.account.log("imap-direct: calling idle_check")
-        res = self.conn.idle_check(timeout=30)
-        if len(res) == 0:
-            raise TimeoutError
+        res = self.conn.idle_check()
```

Comment:
Shouldn't this still raise a TimeoutError if `len(res) == 0`? Or will `conn.idle_check()` raise an error by itself in this case now?

Reply:
Without timeout it should not return without a response. If it still does, we return empty `res` and outside code will restart `idle_check` if it wants a response.